identity_normalizer: remove otherId dissertationId when it's duplicated by sourceId

ndushay opened the pull request:
## Why was this change made?

Closes #3402

Note that there should only ever be a single dissertationID associated with an ETD object.  

I would be shocked if any non-ETD object had a dissertation id (giggle).

## How was this change tested?

ran `druid:wn922bm5946` through validation and it now passes roundtrip.

The below proves it ... causes no harm??

```
$ bin/validate-cocina-roundtrip -s 100000 -i druids.testbed.txt
```

### THIS BRANCH (with change)

```
Status (n=100000; not using Missing for success/different/error stats):
  Success:   97171 (97.251%)
  Different: 2747 (2.749%)
  Mapping error:     0 (0.0%)
  Create error:     0 (0.0%)
  Normalization error:     0 (0.0%)
  Missing from cache:     0 (0.0%)
  Unmapped:     0 (0.0%)
  Expected unmapped:     82 (0.082%)
  Bad cache:     0 (0.0%)
```

### MAIN BRANCH (before change)

```
Status (n=100000; not using Missing for success/different/error stats):
  Success:   97171 (97.251%)
  Different: 2747 (2.749%)
  Mapping error:     0 (0.0%)
  Create error:     0 (0.0%)
  Normalization error:     0 (0.0%)
  Missing from cache:     0 (0.0%)
  Unmapped:     0 (0.0%)
  Expected unmapped:     82 (0.082%)
  Bad cache:     0 (0.0%)
```


## Which documentation and/or configurations were updated?

mjgiarlo reviewed (commented):
How does this change relate to this other normalization method? https://github.com/sul-dlss/dor-services-app/pull/3410/files#diff-580d483ef8eddd756014c0969e83e029f7fe7100a7e3ca251de4437dd1c6b6b3R64-R77 Intended to have both?

mjgiarlo commented on app/services/cocina/normalizers/identity_normalizer.rb:
```diff
@@ -161,14 +162,24 @@ def normalize_object_creator
       end
+
+      def normalize_otherid_dissertationid
+        return if ng_xml.root.xpath('//otherId[@name="dissertationid"]').blank?
+
+        source_id_node = ng_xml.root.xpath('//sourceId[@source="dissertation"]').first
+        return if source_id_node.blank?
+
+        other_id = ng_xml.root.xpath('//otherId[@name="dissertationid"]').first
+        ng_xml.root.xpath('//otherId[@name="dissertationid"]').each(&:remove) if source_id_node.text == other_id.text
```
I wonder if we can reduce the mental load of what this method is doing by considering something like the following:

```ruby
        other_id_node, source_id_node = ng_xml.root.xpath('//otherId[@name="dissertationid"] | //sourceId[@source="dissertation"]')
        return if other_id_node.blank? || source_id_node.blank?

        other_id_node.remove if source_id_node.text == other_id_node.text
```

❓

ndushay replied:
Great suggestion, Mike.  I made one tweak for reducing my own mental load.   And for some reason, I was hitting a bug when I tried `.remove` on a single node, but I must've had another bug at the same time.

mjgiarlo reviewed (commented):
I said this elsewhere, but to tie it directly to a review: I'd be interested in the relationship between the method you added and this other existing method which seems to be touching the same two data structures: https://github.com/sul-dlss/dor-services-app/blob/d81397f5bb9092c0e079fc2938f9ff022eb7868b/app/services/cocina/normalizers/identity_normalizer.rb#L67-L78

ndushay commented on app/services/cocina/normalizers/identity_normalizer.rb:
```diff
@@ -62,21 +63,6 @@ def normalize_source_id_whitespace
         end
       end
 
-      # if there is no sourceId, but there is an otherId with name dissertationid, convert the dissertationid to a sourceId
-      #   (obviously only applies to (old) ETDs)
-      def normalize_dissertation_id_to_source_id
-        return if ng_xml.root.xpath('//sourceId').present?
-
-        diss_id_nodes = ng_xml.root.xpath('//otherId[@name="dissertationid"]')
-        return if diss_id_nodes.first&.content.blank?
-
-        new_source_id_node = Nokogiri::XML::Node.new('sourceId', ng_xml)
-        new_source_id_node.content = diss_id_nodes.first.content
-        new_source_id_node['source'] = 'dissertationid'
-        diss_id_nodes.first.parent.add_child(new_source_id_node)
-        diss_id_nodes.first.remove
-      end
-
```
moved to be close to its sibling method, newly added.

ndushay commented on app/services/cocina/normalizers/identity_normalizer.rb:
```diff
@@ -172,6 +158,30 @@ def normalize_object_label
+
+      # if there is no sourceId and there is an otherId[@name="dissertationid"], convert the otherId to a sourceId
+      #   (only applies to (old) ETDs)
+      def add_missing_sourceid_from_otherid_dissertationid
+        return if ng_xml.root.xpath('//sourceId').present?
+        # NOTE: it has been empirically proven that there are no empty elements for these
+        return if ng_xml.root.xpath('//otherId[@name="dissertationid"]').blank?
+
+        diss_id_node = ng_xml.root.xpath('//otherId[@name="dissertationid"]').first
+        diss_id_node.node_name = 'sourceId'
+        diss_id_node['source'] = diss_id_node['name']
+        diss_id_node.delete('name')
+      end
```
a refactored version of `normalize_dissertation_id_to_source_id` with a clearer name and more grokkable code.

ndushay reviewed (commented):
ok Mike - have at it.

ndushay commented:
> I said this elsewhere, but to tie it directly to a review: I'd be interested in the relationship between the method you added and this other existing method which seems to be touching the same two data structures:
> 
> https://github.com/sul-dlss/dor-services-app/blob/main/app/services/cocina/normalizers/identity_normalizer.rb#L67-L78

Mike and I conferred and agreed the methods were different and some of the confusion was from poor naming and code that was a bit tortured.  I renamed both my new method and the other method and refactored the code to make it clearer.
